vip-cache-manager/js/admin-bar.js

```diff
@@ -0,0 +1,69 @@
+(async () => {
+	async function postData(url = '', data = {}) {
+		// Default options are marked with *
+		const response = await fetch(url, {
+			method: 'POST',
+			credentials: 'same-origin',
+			headers: {
+				'Content-Type': 'application/json'
+			},
+			body: JSON.stringify({ ...data })
+		});
+		return response.json();
+	}
+
+	// Indicates whether there's a purge request happening
+	let purgeQueried = false;
+	// Stores the ref to the DOM node
+	let btn;
+
+	/**
+	 * Grab all the necesary URLs (incl. scripts and CSS) for the purge.
+	 */
+	const getURLsToPurge = () => {
+		return [document.location.toString()].concat(
+			Array.from(document.querySelectorAll('script[src]')).map(({ src }) => src),
+			Array.from(document.querySelectorAll('link[rel=stylesheet]')).map(({ href }) => href)
+		)
+			.filter(url => url.includes(document.location.hostname));
+	}
+
+	/**
+	 * Cache purge click handler.
+	 * 
+	 * @param {Event} e
+	 */
+	async function onClickHandler(e) {
+		if (purgeQueried) {
+			return;
+		}
+
+		let { nonce = '', ajaxurl = '' } = window.VIPPageFlush || {};
+
+		if (!(nonce && ajaxurl)) {
+			alert('VIP Cache Manager: page cache purging disabled');
+		}
+
+		purgeQueried = true;
+
+		const urls = getURLsToPurge();
+
+		try {
+			const res = await postData(`${ajaxurl}`, { nonce, urls });
```

Comment:
Unclear why we need a string literal there:

```suggestion
			const res = await postData(ajaxurl, { nonce, urls });
```